### ISSUE TITLE
inference crashing on recently trained models

### ISSUE THREAD

**Issue by initze** [initze/thaw-slump-segmentation#19]
inference on recently trained models causes weird crash with some printout of model details

on older models, models are running OK. I'll check at what pount that issue started to occur

`an", "module.encoder.layer4.2.bn2.running_var", "module.encoder.layer4.2.bn2.num_batches_tracked", "module.decoder.blocks.x_0_0.conv1.0.weight", "module.decoder.blocks.x_0_0.conv1.1.weight", "module.decoder.blocks.x_0_0.conv1.1.bias", "module.decoder.blocks.x_0_0.conv1.1.running_mean", "module.decoder.blocks.x_0_0.conv1.1.running_var", "module.decoder.blocks.x_0_0.conv1.1.num_batches_tracked", "module.decoder.blocks.x_0_0.conv2.0.weight", "module.decoder.blocks.x_0_0.conv2.1.weight", "module.decoder.blocks.x_0_0.conv2.1.bias", "module.decoder.blocks.x_0_0.conv2.1.running_mean", "module.decoder.blocks.x_0_0.conv2.1.running_var", "module.decoder.blocks.x_0_0.conv2.1.num_batches_tracked", "module.decoder.blocks.x_0_1.conv1.0.weight", "module.decoder.blocks.x_0_1.conv1.1.weight", "module.decoder.blocks.x_0_1.conv1.1.bias", "module.decoder.blocks.x_0_1.conv1.1.running_mean", "module.decoder.blocks.x_0_1.conv1.1.running_var", "module.decoder.blocks.x_0_1.conv1.1.num_batches_tracked", "module.decoder.blocks.x_0_1.conv2.0.weight", "module.decoder.blocks.x_0_1.conv2.1.weight", "module.decoder.blocks.x_0_1.conv2.1.bias", "module.decoder.blocks.x_0_1.conv2.1.running_mean", "module.decoder.blocks.x_0_1.conv2.1.running_var", "module.decoder.blocks.x_0_1.conv2.1.num_batches_tracked", "module.decoder.blocks.x_1_1.conv1.0.weight", "module.decoder.blocks.x_1_1.conv1.1.weight", "module.decoder.blocks.x_1_1.conv1.1.bias", "module.decoder.blocks.x_1_1.conv1.1.running_mean", "module.decoder.blocks.x_1_1.conv1.1.running_var", "module.decoder.blocks.x_1_1.conv1.1.num_batches_tracked", "module.decoder.blocks.x_1_1.conv2.0.weight", "module.decoder.blocks.x_1_1.conv2.1.weight", "module.decoder.blocks.x_1_1.conv2.1.bias", "module.decoder.blocks.x_1_1.conv2.1.running_mean", "module.decoder.blocks.x_1_1.conv2.1.running_var", "module.decoder.blocks.x_1_1.conv2.1.num_batches_tracked", "module.decoder.blocks.x_0_2.conv1.0.weight", "module.decoder.blocks.x_0_2.conv1.1.weight", "module.decoder.blocks.x_0_2.conv1.1.bias", "module.decoder.blocks.x_0_2.conv1.1.running_mean", "module.decoder.blocks.x_0_2.conv1.1.running_var", "module.decoder.blocks.x_0_2.conv1.1.num_batches_tracked", "module.decoder.blocks.x_0_2.conv2.0.weight", "module.decoder.blocks.x_0_2.conv2.1.weight", "module.decoder.blocks.x_0_2.conv2.1.bias", "module.decoder.blocks.x_0_2.conv2.1.running_mean", "module.decoder.blocks.x_0_2.conv2.1.running_var", "module.decoder.blocks.x_0_2.conv2.1.num_batches_tracked", "module.decoder.blocks.x_1_2.conv1.0.weight", "module.decoder.blocks.x_1_2.conv1.1.weight", "module.decoder.blocks.x_1_2.conv1.1.bias", "module.decoder.blocks.x_1_2.conv1.1.running_mean", "module.decoder.blocks.x_1_2.conv1.1.running_var", "module.decoder.blocks.x_1_2.conv1.1.num_batches_tracked", "module.decoder.blocks.x_1_2.conv2.0.weight", "module.decoder.blocks.x_1_2.conv2.1.weight", "module.decoder.blocks.x_1_2.conv2.1.bias", "module.decoder.blocks.x_1_2.conv2.1.running_mean", "module.decoder.blocks.x_1_2.conv2.1.running_var", "module.decoder.blocks.x_1_2.conv2.1.num_batches_tracked", "module.decoder.blocks.x_2_2.conv1.0.weight", "module.decoder.blocks.x_2_2.conv1.1.weight", "module.decoder.blocks.x_2_2.conv1.1.bias", "module.decoder.blocks.x_2_2.conv1.1.running_mean", "module.decoder.blocks.x_2_2.conv1.1.running_var", "module.decoder.blocks.x_2_2.conv1.1.num_batches_tracked", "module.decoder.blocks.x_2_2.conv2.0.weight", "module.decoder.blocks.x_2_2.conv2.1.weight", "module.decoder.blocks.x_2_2.conv2.1.bias", "module.decoder.blocks.x_2_2.conv2.1.running_mean", "module.decoder.blocks.x_2_2.conv2.1.running_var", "module.decoder.blocks.x_2_2.conv2.1.num_batches_tracked", "module.decoder.blocks.x_0_3.conv1.0.weight", "module.decoder.blocks.x_0_3.conv1.1.weight", "module.decoder.blocks.x_0_3.conv1.1.bias", "module.decoder.blocks.x_0_3.conv1.1.running_mean", "module.decoder.blocks.x_0_3.conv1.1.running_var", "module.decoder.blocks.x_0_3.conv1.1.num_batches_tracked", "module.decoder.blocks.x_0_3.conv2.0.weight", "module.decoder.blocks.x_0_3.conv2.1.weight", "module.decoder.blocks.x_0_3.conv2.1.bias", "module.decoder.blocks.x_0_3.conv2.1.running_mean", "module.decoder.blocks.x_0_3.conv2.1.running_var", "module.decoder.blocks.x_0_3.conv2.1.num_batches_tracked", "module.decoder.blocks.x_1_3.conv1.0.weight", "module.decoder.blocks.x_1_3.conv1.1.weight", "module.decoder.blocks.x_1_3.conv1.1.bias", "module.decoder.blocks.x_1_3.conv1.1.running_mean", "module.decoder.blocks.x_1_3.conv1.1.running_var", "module.decoder.blocks.x_1_3.conv1.1.num_batches_tracked", "module.decoder.blocks.x_1_3.conv2.0.weight", "module.decoder.blocks.x_1_3.conv2.1.weight", "module.decoder.blocks.x_1_3.conv2.1.bias", "module.decoder.blocks.x_1_3.conv2.1.running_mean", "module.decoder.blocks.x_1_3.conv2.1.running_var", "module.decoder.blocks.x_1_3.conv2.1.num_batches_tracked", "module.decoder.blocks.x_2_3.conv1.0.weight", "module.decoder.blocks.x_2_3.conv1.1.weight", "module.decoder.blocks.x_2_3.conv1.1.bias", "module.decoder.blocks.x_2_3.conv1.1.running_mean", "module.decoder.blocks.x_2_3.conv1.1.running_var", "module.decoder.blocks.x_2_3.conv1.1.num_batches_tracked", "module.decoder.blocks.x_2_3.conv2.0.weight", "module.decoder.blocks.x_2_3.conv2.1.weight", "module.decoder.blocks.x_2_3.conv2.1.bias", "module.decoder.blocks.x_2_3.conv2.1.running_mean", "module.decoder.blocks.x_2_3.conv2.1.running_var", "module.decoder.blocks.x_2_3.conv2.1.num_batches_tracked", "module.decoder.blocks.x_3_3.conv1.0.weight", "module.decoder.blocks.x_3_3.conv1.1.weight", "module.decoder.blocks.x_3_3.conv1.1.bias", "module.decoder.blocks.x_3_3.conv1.1.running_mean", "module.decoder.blocks.x_3_3.conv1.1.running_var", "module.decoder.blocks.x_3_3.conv1.1.num_batches_tracked", "module.decoder.blocks.x_3_3.conv2.0.weight", "module.decoder.blocks.x_3_3.conv2.1.weight", "module.decoder.blocks.x_3_3.conv2.1.bias", "module.decoder.blocks.x_3_3.conv2.1.running_mean", "module.decoder.blocks.x_3_3.conv2.1.running_var", "module.decoder.blocks.x_3_3.conv2.1.num_batches_tracked", "module.decoder.blocks.x_0_4.conv1.0.weight", "module.decoder.blocks.x_0_4.conv1.1.weight", "module.decoder.blocks.x_0_4.conv1.1.bias", "module.decoder.blocks.x_0_4.conv1.1.running_mean", "module.decoder.blocks.x_0_4.conv1.1.running_var", "module.decoder.blocks.x_0_4.conv1.1.num_batches_tracked", "module.decoder.blocks.x_0_4.conv2.0.weight", "module.decoder.blocks.x_0_4.conv2.1.weight", "module.decoder.blocks.x_0_4.conv2.1.bias", "module.decoder.blocks.x_0_4.conv2.1.running_mean", "module.decoder.blocks.x_0_4.conv2.1.running_var", "module.decoder.blocks.x_0_4.conv2.1.num_batches_tracked", "module.segmentation_head.0.weight", "module.segmentation_head.0.bias".`

**Comment by initze** [comment]
I'll check if the Parallel training mode has something to do with that.

**Comment by initze** [comment]
models trained in 0.4.1 are working

trained in 0.5.2 is crashing

config.yml structure hasn't changed.

Crash occurs here:     [model.load_state_dict(torch.load(ckpt, map_location=dev))](https://github.com/initze/thaw-slump-segmentation/blob/2551b0753e14161a36c7ba00e0bf85243598a8f6/inference.py#L280)

**Comment by initze** [comment]
stupid issue 🤦‍♂️ 🤦‍♂️ 🤦‍♂️ 

fixed with https://github.com/initze/thaw-slump-segmentation/commit/866e7a106687712b46054fe8fd9160c93c4051bb

parallel trained models need to be explicitly declared nn.DataParallel for inference like in training stage. created workaround with try/except
